back-end/API_Documentation/AuthAndUser.md

```diff
@@ -0,0 +1,193 @@
+
+* **Request Method:** POST
+
+* **Body Params:**
+
+    **Required:**
+
+    | Name       | Type   | Description                         |
+    | ---------- | ------ | ----------------------------------- |
+    | `username` | String | The username of the user to create. |
+    | `password` | String | The password of the user to create. |
+
+* **Returned Data Format:** Plain Text
+
+* **Success Response:**
+
+    * **Code:** 200 </br>
+    **Content:** `"Signup Successful"`
+
+* **Error Response:**
+
+    Missing one or more body parameters
+
+    * **Code:** 400 </br>
+    **Content:** `"Missing username or password"`
+
+    Username is taken by another user already
+
+    * **Code:** 400 </br>
+    **Content:** `"Username already taken"`
+
+    </br>
+
+### Logging out user
+
+Deletes authorization cookie.
+
+* **Endpoint:** /logout
+
+* **Request Method:** POST
+
+* **Returned Data Format:** Plain Text
+
+* **Success Response:**
+
+    * **Code:** 200 </br>
+    **Content:** `"Logout Successful"`
+
+    </br>
+
+### Setting a Users Preferred Color Scheme
+
+Sets a user's preferred color scheme. Valid color schemes are either `"light"` or `"dark"` mode.
+* **Endpoint:** /color_scheme
+
+* **Request Method:** POST
+
+* **Body Params:**
+
+    **Required:**
+
+    | Name           | Type   | Description                                       |
+    | -------------- | ------ | ------------------------------------------------- |
+    | `username`     | String | The username of the user to create.               |
+    | `color_scheme` | String | The preferred color scheme for the given user.    |
+
+* **Returned Data Format:** Plain Text
+
+* **Success Response:**
+
+    * **Code:** 200 </br>
+    **Content:** `"Color Scheme Updated Successfuly"`
+
+* **Error Response:**
+
+    Missing one or more body parameters
+
+    * **Code:** 400 </br>
+    **Content:** `"Missing username or color_scheme"`
+
+    Username doesn't match any existing user
+
+    * **Code:** 401 </br>
+    **Content:** `"Invalid username"`
+
+    An invalid color scheme is provided
+
+    * **Code:** 400 </br>
+    **Content:** `"Invalid color scheme"`
+
+    </br>
+
+### Grabbing color_scheme from cookie
+
+Grabs the color_scheme stored as a cookie. Valid color schemes are either `"light"` or `"dark"` mode. Will default to `"light"` if no cookie has been set. For clarification, the sample call is just `/grab/color_scheme`.
+* **Endpoint:** /grab/color_scheme
+
+* **Request Method:** GET
+
+* **Returned Data Format:** JSON
+
+* **Success Response:**
+
+    * **Code:** 200 </br>
+    **Content:**
+
+    ```JSON
+    {
+        "color_scheme": "light"
+    }
+    ```
+
+    </br>
+
+### Grabbing username from cookie
+
+Grabs the username of the user that was previously logged in. Note that username cookies expire after 2 days.
+* **Endpoint:** /grab/username
+
+* **Request Method:** GET
+
+* **Returned Data Format:** JSON
+
+* **Sample Request: `/grab/elliot`**
```

Comment:
The "username" in the endpoint `/grab/username` is just part of the path to access that given endpoint. It is not a parameter and no sample request is needed. In order to use this endpoint, the front-end just needs to visit the `/grab/username` endpoint with a cookie specifying a user. **We should delete this sample request**

Reply:
Patched, please review